diff --git a/plugins/rdp/rdp_event.c b/plugins/rdp/rdp_event.c
index 59f51c9..8787690 100644
--- a/plugins/rdp/rdp_event.c
+++ b/plugins/rdp/rdp_event.c
@@ -877,20 +877,11 @@ static gboolean remmina_rdp_event_on_key(GtkWidget *widget, GdkEventKey *event,
 					}
 				}
 			}
-#if FREERDP_CHECK_VERSION(3, 11, 0)			
-			guint32 keyboard_type = freerdp_settings_get_uint32(rfi->clientContext.context.settings, FreeRDP_KeyboardType);
-			if (keyboard_type == 0){
-				keyboard_type = WINPR_KBD_TYPE_IBM_ENHANCED;
-			}
-#ifdef GDK_WINDOWING_X11
-			DWORD vc = GetVirtualKeyCodeFromKeycode(hardware_keycode, WINPR_KEYCODE_TYPE_XKB);
-#else
-			DWORD vc = GetVirtualKeyCodeFromKeycode(hardware_keycode, WINPR_KEYCODE_TYPE_EVDEV);
-#endif
-			const DWORD sc = GetVirtualScanCodeFromVirtualKeyCode(vc, keyboard_type); 
+			DWORD sc = freerdp_keyboard_get_rdp_scancode_from_x11_keycode(hardware_keycode);
+#if FREERDP_CHECK_VERSION(3, 11, 0)
 			DWORD scancode = freerdp_keyboard_remap_key(rfi->remap_table, sc);
 #else
-			scancode = freerdp_keyboard_get_rdp_scancode_from_x11_keycode(hardware_keycode);
+			scancode = sc;
 #endif
 			if (scancode) {
 				rdp_event.key_event.key_code = scancode & 0xFF;
@@ -915,21 +906,11 @@ static gboolean remmina_rdp_event_on_key(GtkWidget *widget, GdkEventKey *event,
 			    unicode_keyval == 0 ||                                                      // Impossible to translate
 			    (event->state & (GDK_MOD1_MASK | GDK_CONTROL_MASK | GDK_SUPER_MASK)) != 0   // A modifier not recognized by gdk_keyval_to_unicode()
 			    ) {
+				DWORD sc = freerdp_keyboard_get_rdp_scancode_from_x11_keycode(event->hardware_keycode);
 #if FREERDP_CHECK_VERSION(3, 11, 0)
-
-				guint32 keyboard_type = freerdp_settings_get_uint32(rfi->clientContext.context.settings, FreeRDP_KeyboardType);
-				if (keyboard_type == 0){
-					keyboard_type = WINPR_KBD_TYPE_IBM_ENHANCED;
-				}
-#ifdef GDK_WINDOWING_X11
-				DWORD vc = GetVirtualKeyCodeFromKeycode(hardware_keycode, WINPR_KEYCODE_TYPE_XKB);
-#else
-				DWORD vc = GetVirtualKeyCodeFromKeycode(hardware_keycode, WINPR_KEYCODE_TYPE_EVDEV);
-#endif
-				const DWORD sc = GetVirtualScanCodeFromVirtualKeyCode(vc, keyboard_type); 
 				DWORD scancode = freerdp_keyboard_remap_key(rfi->remap_table, sc);
 #else
-				scancode = freerdp_keyboard_get_rdp_scancode_from_x11_keycode(event->hardware_keycode);
+				scancode = sc;
 #endif
 				rdp_event.key_event.key_code = scancode & 0xFF;
 				rdp_event.key_event.extended = scancode & 0x100;
